molecule/default/tests/test_default.py

```diff
@@ -15,33 +15,35 @@
 @pytest.mark.parametrize("pkg", ["apt-transport-https"])
 def test_apt_https_support(host, pkg):
     """Ensure apt has support for repository URLs using HTTPS."""
-    assert host.package(pkg).is_installed
+    if host.system_info.distribution in ["debian", "kali", "ubuntu"]:
+        assert host.package(pkg).is_installed
 
 
 def test_source_list_for_http(host):
     """Check if the source list file has any URLs still using HTTP."""
-    source_file = "/etc/apt/sources.list"
-    # As of Debian Bullseye the /etc/apt/sources.list file has moved
-    # to /etc/apt/sources.list.d/debian.sources
-    if (
-        host.system_info.distribution == "debian"
-        and host.system_info.codename == "bookworm"
-    ):
-        source_file = "/etc/apt/sources.list.d/debian.sources"
-
-    file_lines = host.file(source_file).content_string.split(os.linesep)
-    sources = [line for line in file_lines if line.startswith("deb")]
-
-    for source in sources:
-        source_elements = source.split(" ")
-        # The second element of a source entry can optionally be a set of options
-        # in the form "[options]".
-        repository_url = source_elements[
-            1 if not source_elements[1].startswith("[") else 2
-        ]
-        if host.system_info.distribution == "ubuntu" and (
-            "ubuntu.com" in repository_url or "canonical.com" in repository_url
+    if host.system_info.distribution in ["debian", "kali", "ubuntu"]:
+        source_file = "/etc/apt/sources.list"
+        # As of Debian Bullseye the /etc/apt/sources.list file has
```

Comment:
There's a comment about Debian Bullseye moving the sources list to a new location, but you're only checking the new location for Debian Bookworm. Consider if this should apply to Bullseye and newer versions.

Reply:
I believe this comment is incorrect and the change is happening with Debian Bookworm (as tested). The Docker images seem to support this:

```console
$ docker run -it --rm debian:buster ls -l /etc/apt /etc/apt/sources.list.d
/etc/apt:
total 24
drwxr-xr-x 2 root root 4096 Sep 12  2022 apt.conf.d
drwxr-xr-x 2 root root 4096 Apr 19  2021 auth.conf.d
drwxr-xr-x 2 root root 4096 Apr 19  2021 preferences.d
-rw-r--r-- 1 root root  416 Sep 12  2022 sources.list
drwxr-xr-x 2 root root 4096 Apr 19  2021 sources.list.d
drwxr-xr-x 2 root root 4096 Sep 12  2022 trusted.gpg.d

/etc/apt/sources.list.d:
total 0
$ docker run -it --rm debian:bookworm ls -l /etc/apt /etc/apt/sources.list.d
/etc/apt:
total 24
drwxr-xr-x 2 root root 4096 Feb 27 00:00 apt.conf.d
drwxr-xr-x 2 root root 4096 Feb  8 16:07 auth.conf.d
drwxr-xr-x 2 root root 4096 Feb  8 16:07 keyrings
drwxr-xr-x 2 root root 4096 Feb  8 16:07 preferences.d
drwxr-xr-x 2 root root 4096 Feb 27 00:00 sources.list.d
drwxr-xr-x 2 root root 4096 Feb 27 00:00 trusted.gpg.d

/etc/apt/sources.list.d:
total 4
-rw-r--r-- 1 root root 443 Feb 27 00:00 debian.sources
```

Reply:
This is also supported by the fact that only Bookworm comes with a 2.x version of `apt`.